tools/benchmark/src/mocha/memoryTestRunner.ts

```diff
@@ -296,6 +304,16 @@ export function benchmarkMemory(testObject: IMemoryTestObject): Test {
 						heapUsedStats.marginOfErrorPercent > args.maxRelativeMarginOfError
 					);
 
+					const avgHeapUsed = heapUsedStats.arithmeticMean;
+					const allowedMemoryUsage =
+						args.baselineMemoryUsage * (1 + ALLOWED_DEVIATION / 100);
+
+					if (avgHeapUsed > allowedMemoryUsage) {
+						throw new Error(
+							`Memory Regression detected for ${testObject.title}: Used ${avgHeapUsed} bytes, exceeding the baseline of ${allowedMemoryUsage} bytes.`,
```

Comment:
Nit: `allowedMemoryUsage` isn't the actual baseline, it's the baseline + the allowed deviation. It might be better to express this like
```
							`Memory Regression detected for ${testObject.title}: Used ${avgHeapUsed} bytes, exceeding the baseline of ${args.baselineMemoryUsage} bytes., with an allowed tolerance of {tolerance}.`,
```

Reply:
Good catch, will fix!